### DISCUSSION TITLE
Convert Ecore to OML

### DISCUSSION THREAD

**Discussion by alfa-ryano** [opencaesar/platform#21] [General]
Hi everyone,

Is there any available tool to convert Ecore metamodel to OML? 
I have checked a tool here: https://github.com/Daylyt247/ecore-adapter/tree/master/ecore2oml
but apparently, it doesn't work. It shows me the following error:

'could not resolve http://www.w3.org/2001/XMLSchema#string in context of ...'

Kind regards,

**Comment by melaasar** [comment]
This is usually a symptom of not having the core-vocabularies in your project's gradle dependencies or not having the proper URI mapping entry in the project's catalog.xml file. 

I recommend starting by creating an OML project using an oml-template (https://github.com/opencaesar/oml-template). Then,
- Remove the OML ontologies in `src/oml` folder (you can create a vocabulary bundle later). Add the ecore files in folder `src/ecore`.
- Add/configure the ecore2oml task in your build.folder (https://github.com/opencaesar/ecore-adapter/blob/master/ecore2oml/README.md). Choose the output folder as `build/oml`.
- Then run with `./gradlew ecore2oml`, or run the Ecore2Oml App from Eclipse giving the command arguments as params.

Notice that we have recently updated the (https://github.com/opencaesar/ecore-adapter) repository to support OML v2. It also includes improvement for the ecore2oml transformation. So pull that or sync your fork with this and try again.

<details>
  <summary>If you have already created an OML 1.x project, then...</summary>

   you will need to upgrade it to oml 2. You can do the following:

   - In build.gradle, under buildScript, change the tool dependencies to 2.+, for example:

   ```
    buildscript {
       dependencies {
           classpath 'io.opencaesar.owl:owl-fuseki-gradle:2.+'
           classpath 'io.opencaesar.owl:owl-shacl-fuseki-gradle:2.+'
           classpath 'io.opencaesar.owl:owl-query-gradle:2.+'
           classpath 'io.opencaesar.owl:owl-load-gradle:2.+'
           classpath 'io.opencaesar.owl:owl-reason-gradle:2.+'
           classpath 'io.opencaesar.owl:owl-doc-gradle:2.+'
           classpath 'io.opencaesar.oml:oml-merge-gradle:2.+'
           classpath 'io.opencaesar.adapters:oml2owl-gradle:2.+'
       }
   }
   ```
   - In build.gradle, change the dependencies on core-vocabularies or imce-vocabularies to 4.+. For example,

   ```
   dependencies {
       oml "io.opencaesar.ontologies:core-vocabularies:4.+"
   }
   ```
</details>

Let me know if you have a question.

**Comment by alfa-ryano** [comment]
Thanks for the response. I'm gonna try using the tool. 共 ありがと.
